Tests/GRPCTests/AsyncClientCallTests.swift

```diff
@@ -0,0 +1,232 @@
+      XCTAssertNoThrow(try group.syncShutdownGracefully())
+    }
+
+    super.tearDown()
+  }
+
+  func testAsyncUnaryCall() throws { XCTAsyncTest {
+    let channel = try self.setUpServerAndChannel()
+    let get: AsyncUnaryCall<Echo_EchoRequest, Echo_EchoResponse> = channel.makeAsyncUnaryCall(
+      path: "/echo.Echo/Get",
+      request: .with { $0.text = "get" },
+      callOptions: .init()
+    )
+
+    await assertThat(try await get.response, .doesNotThrow())
+    await assertThat(await get.status, .hasCode(.ok))
+  } }
+
+  func testAsyncClientStreamingCall() throws { XCTAsyncTest {
+    let channel = try self.setUpServerAndChannel()
+    let collect: AsyncClientStreamingCall<Echo_EchoRequest, Echo_EchoResponse> = channel
+      .makeAsyncClientStreamingCall(
+        path: "/echo.Echo/Collect",
+        callOptions: .init()
+      )
+
+    for word in ["boyle", "jeffers", "holt"] {
+      try await collect.sendMessage(.with { $0.text = word })
+    }
+    try await collect.sendEnd()
+
+    await assertThat(try await collect.response, .doesNotThrow())
+    await assertThat(await collect.status, .hasCode(.ok))
+  } }
+
+  func testAsyncServerStreamingCall() throws { XCTAsyncTest {
+    let channel = try self.setUpServerAndChannel()
+    let expand: AsyncServerStreamingCall<Echo_EchoRequest, Echo_EchoResponse> = channel
+      .makeAsyncServerStreamingCall(
+        path: "/echo.Echo/Expand",
+        request: .with { $0.text = "boyle jeffers holt" },
+        callOptions: .init()
+      )
+
+    var numResponses = 0
+    for try await _ in expand.responseStream {
+      numResponses += 1
+    }
+    await assertThat(numResponses, .is(.equalTo(3)))
+    await assertThat(await expand.status, .hasCode(.ok))
+  } }
+
+  func testAsyncBidirectionalStreamingCall() throws { XCTAsyncTest {
+    let channel = try self.setUpServerAndChannel()
+    let update: AsyncBidirectionalStreamingCall<Echo_EchoRequest, Echo_EchoResponse> = channel
+      .makeAsyncBidirectionalStreamingCall(
+        path: "/echo.Echo/Update",
+        callOptions: .init()
+      )
+
+    for word in ["boyle", "jeffers", "holt"] {
+      try await update.sendMessage(.with { $0.text = word })
+    }
+    try await update.sendEnd()
+
+    var numResponses = 0
+    for try await _ in update.responseStream {
+      numResponses += 1
+    }
+    await assertThat(numResponses, .is(.equalTo(3)))
+    await assertThat(await update.status, .hasCode(.ok))
+  } }
+
+  func testAsyncBidirectionalStreamingCall_InterleavedRequestsAndResponses() throws { XCTAsyncTest {
+    let channel = try self.setUpServerAndChannel()
+    let update: AsyncBidirectionalStreamingCall<Echo_EchoRequest, Echo_EchoResponse> = channel
+      .makeAsyncBidirectionalStreamingCall(
+        path: "/echo.Echo/Update",
+        callOptions: .init()
+      )
+
+    // Spin up a task to send the requests with a delay before each one
+    Task {
+      let delay = TimeAmount.milliseconds(500)
```

Comment:
Rather than introducing a delay can we use an iterator from the response stream?

```swift
let responseStreamIterator = update.responseStream.makeIterator()
for word in ["foo", "bar", "bar"] {
  try await update.sendMessage(...)
  _ = try await responseStreamIterator.next()
  numResponses += 1
}
```